include/envoy/server/admin.h

```diff
@@ -15,6 +16,39 @@
 namespace Envoy {
 namespace Server {
 
+class AdminStream {
+public:
+  virtual ~AdminStream() {}
+
+  /**
+   * @param end_stream set to false for streaming response. Default is true, which will
+   * end the response when the initial handler completes.
+   */
+  virtual void setEndStreamOnComplete(bool end_stream) PURE;
+
+  /**
+   * @param cb callback to be added to the list of callbacks invoked by onDestroy() when connection
+   * is closed.
+   */
+  virtual void addOnDestroyCallback(std::function<void()> cb) PURE;
```

Comment:
nit/taste: I feel like the 'On' here and the 'Callback' both add the same info, and it would be slightly nicer to call this 'addDestroyCallback', but I'm fine as it is too.

Reply:
thanks for the comment. I prefer to keep the name since it references the onDestroy() method, where the callbacks are being used.